### ISSUE TITLE
Update Items endpints

### ISSUE THREAD

**Issue by emilymdubois** [osmlab/to-fix-backend#143]
We're going to use [both POST and PUT operations](https://github.com/osmlab/to-fix-backend/issues/135), so we need to update how we create projects.

- [x] Allow user to provide item `id` for URL queries, but assign an auto-generated ID on the backend
- [x] Add a `POST /projects/:project/items` endpoint
- [x] Update the `PUT /projects/:project/items/:item` endpoint to exclude creation events

Similar to [tags](https://github.com/osmlab/to-fix-backend/issues/140), we'll want users to be able to pass an optional query parameter to filter projects by name instead of ID. Below, all instances of `:item` can be assumed to mean item ID, and all instances of `item` can be assumed to mean item name.

@batpad @kepta @samanpwbb

---

## Get all project items
```sh
GET /projects/:project/items
```
#### Query parameters
Name | Type | Description
---|---|---
`item` | string | **Optional.** May be passed to filter returned items where item name contains the provided string.

## Create an item
```sh
POST /projects/:project/items
```
#### Payload parameters
Name | Type | Description
---|---|---
`name` | string | **Required.** The item name.
`instructions` | string | **Required.** The instructions on how to work on the item.
`pin` | string | **Required.** A 2D geometry point to represent the feature.
`status` | enum<string> | **Optional.** The status of ticket. Defaults to `open`.
`featureCollection` | object | **Optional.** A GeoJSON context feature.
`lock` | enum<string> | **Optional.** Item's locked status.

## Get an item
```sh
GET /projects/:project/items/:item
```

## Update an item
```sh
PUT /projects/:project/items/:item
```
#### Payload parameters
Name | Type | Description
---|---|---
`name` | string | **Optional.** The item name.
`instructions` | string | **Optional.** The instructions on how to work on the item.
`pin` | string | **Optional.** A 2D geometry point to represent the feature.
`status` | enum<string> | **Optional.** The status of ticket. Defaults to `open`.
`featureCollection` | object | **Optional.** A GeoJSON context feature.
`lock` | enum<string> | **Optional.** Item's locked status.

## Delete an item
```sh
DELETE /projects/:project/items/:item
```

**Comment by batpad** [comment]
> Add name to the item schema since id will now be an auto-generated integer

@emilymdubois if the `id` is ever going to be user facing, for eg. in a URL, I'd strongly advocate for `uuid`s and not auto incrementing integers. Happy to talk through.

**Comment by emilymdubois** [comment]
@batpad yes, I think we could keep the `auto_id` as the primary but masked ID, and then add an immutable field on a create operation that the user would use to in URL paths. PagerDuty always uses a 6-character string that always starts with `P`.

**Comment by samanpwbb** [comment]
Closed by https://github.com/osmlab/to-fix-backend/pull/162
